### ISSUE TITLE
Write back tagged faces to photo files

### ISSUE THREAD

**Issue by totti4ever** [LibrePhotos/librephotos#130]
Hey, I am still impressed by this piece of of software :-)

Though, even with using LibrePhotos I would still no see an easy way of having faces tagged in pictures over software bounderies, i.e. on mobile and on PC. The interface for tagging faces and of course the supporting algorithm of derived faces would be a perfect base to just write them back to the file, which then could be used on my mobile app again (synched by Syncthing)

**Comment by el-tiuri** [comment]
Was wondering if I should add another issue for this, but decided to piggyback off this; it'd also be fantastic of LibrePhotos could read existing face tags in images. I have a collection of about 100k photos with probably about 20k tagged faces, and it'd be fantastic if this could be transferred over to LibrePhotos.

**Comment by derneuere** [comment]
Is there a general format for saving faces to images? I could support reading from the future LibrePhotos format of saving faces.

**Comment by el-tiuri** [comment]
If I'm understanding it correctly Image Region Metadata is part of both the XMP and IPTC spec (https://web.archive.org/web/20171214222138/http://www.metadataworkinggroup.com/pdf/mwg_guidance.pdf, section 5.9 and https://iptc.org/news/announcing-the-iptc-photo-metadata-standard-2019-1-introducing-iptc-image-regions/ respectively), one application of which is saving faces to images.

**Comment by mcwieger** [comment]
Fully support this request! I have many (manually) tagged photos and it would be great to:

1. read the face tags that are already in the pictures
2. transfer these tags to the LP database (for speed)
3. write the face tags generated by LP AI to the picture (for easier tagging, taking away some of the manual work)

Three applications that do 1. are https://github.com/whatever4711/picapport, https://github.com/bpatrik/pigallery2 and https://github.com/photostructure/photostructure-for-servers.

**Comment by rhatguy** [comment]
I'm adding to this request.  There are so many tools that have some form of face detection now.  I've found that librephotos face detection is one of the best!  BUT, I have an existing gallery of 28k photos that have (mostly) all been tagged through picasa/digikam and the faces have been written into the exif metadata of the pictures.  Not being able to leverage that existing data is a complete show stopper for me.  I can't justify dedicating the time to go back through and re-create all that data that already exists in my photos (even though the face tagging process works great in librephotos).  I've settled on using pigallery2 for now as it reads all my existing tags and leverages the existing data (it doesn't have any face tagging ability on its own).

Secondarily I decided a while back that writing the metadata to the images (or a sidecar file) is a hard requirement for me to use any sort of metadata creation software (face tagging/GPS tagging/keyword creation).  Having been previously a user of what is now called "Gallery the Revival" all the way back to gallery2 and having moved through multiple gallery softwares over time, I've come to understand that gallery softwares come and go, but the pictures stay forever.  Therefore given the time investment I've put into creating that metadata I have a hard requirement for the software to leverage that metadata (and update it if I'm going to use the tool to generate new metadata).

I would encourage the librephotos project to focus on leveraging (and storing) metadata in the image (or in a sidecar).  So many tools today are trying to build new ways of generating metadata but don't focus on the long term curation/leverage of the existing data.

Thanks for the excellent work!

**Comment by Pimmetje** [comment]
https://github.com/python-xmp-toolkit/python-xmp-toolkit

**Comment by rhatguy** [comment]
I see that the ability to write ratings back to photos has recently been added.  Will some of that similar workflow eventually be able to read and write faces to image metadata?

**Comment by alfwro13** [comment]
yep I agree with all of the above, any software that does not leverage already existing data is a show stopper for me. I have already most of my data tagged in one way or another and I cannot use that it's bit pointless doing the same job again.
This project looks very promising. Well done unfortunately for me my search for a photo gallery/manager continues :(
